controllers/topics/updateTopic.js

```diff
@@ -0,0 +1,60 @@
+const {Topics} = require('../../databases/models');
+
+module.exports = async (req, res) => {
+  try {
+    const {icon, cover} = req.body;
+    let {topicName} = req.params;
+    topicName = topicName.toLowerCase();
+    let response;
+
+    let topics = await Topics.findOne({
+      where: {
+        name: topicName,
+        deleted_at: null
+      }
+    });
+
+    if (!topics) {
+      response = {
+        success: false,
+        message: 'Community name not found'
+      };
+
+      return res.status(400).json(response);
+    }
+
+    if (!icon && !cover) {
+      response = {
+        code: 400,
+        success: false,
+        message: 'Please provide icon or cover'
+      };
+
+      return res.status(400).json(response);
+    }
+
+    let updateTopicData = {};
+    if (icon) {
+      updateTopicData.icon_path = icon;
+    }
+    if (cover) {
+      updateTopicData.cover_path = cover;
+    }
+
+    await Topics.update(updateTopicData, {where: {topic_id: topics.topic_id}});
+
+    response = {
+      success: true,
+      message: 'success updated topic details'
+    };
+
+    return res.status(200).json(response);
+  } catch (error) {
+    return res.json({
+      code: error.statusCode,
+      status: 'fail',
+      message: error.message,
+      data: 'null'
+    });
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code looks good overall, but there's a potential issue with error handling. In the catch block, you're assuming that `error` will always have a `statusCode` property. However, not all errors will necessarily have this property (for example, Sequelize errors or JavaScript runtime errors), which could lead to `undefined` being returned as the status code. It would be better to check if `error.statusCode` exists and, if not, return a default error code.

```diff
-      code: error.statusCode,
+      code: error.statusCode || 500,
```
Also, it's not a good practice to send raw error messages to the client because they might contain sensitive information. Instead, consider logging the error message and sending a generic error message to the client.

```diff
-      message: error.message,
+      message: 'An error occurred while updating the topic',
```

Lastly, returning 'null' as a string is not a good practice. If there's no data to return, it's better to omit the `data` field entirely or return `null` without quotes.

```diff
-      data: 'null'
+      data: null
```
This way, the client can check for `null` instead of the string 'null'.


<!-- This is an auto-generated comment by OSS CodeRabbit -->